src/endpoints/onboarding_endpoints.py

```diff
@@ -23,11 +26,83 @@ def __init__(self, client: Steamship, agent_service: AgentService):
         self.client = client
         self.agent_service = agent_service
 
+    @post("/onboarding/character/name")
+    def set_character_name(self, name: str, context_id: Optional[str] = None, **kwargs):
+        context = self.agent_service.build_default_context(
+            context_id=context_id, **kwargs
+        )
+
+        if not _is_allowed_by_moderation(name, context):
+            raise SteamshipError(
+                "Supplied 'name' was rejected by game's moderation filter. Please try again."
+            )
+
+        game_state = get_game_state(context)
+        game_state.player.name = name
+        save_game_state(game_state, context)
+
+    @post("/onboarding/character/background")
+    def set_character_background(
+        self, background: str, context_id: Optional[str] = None, **kwargs
+    ):
+        context = self.agent_service.build_default_context(
+            context_id=context_id, **kwargs
+        )
+
+        if not _is_allowed_by_moderation(background, context):
+            raise SteamshipError(
+                "Supplied 'background' was rejected by game's moderation filter. Please try again."
+            )
+
+        game_state = get_game_state(context)
+        game_state.player.background = background
+        save_game_state(game_state, context)
+
+    @post("/onboarding/character/description")
+    def set_character_description(
+        self,
+        description: str,
+        update: Optional[bool] = True,
+        context_id: Optional[str] = None,
+        **kwargs,
+    ):
+        context = self.agent_service.build_default_context(
+            context_id=context_id, **kwargs
+        )
+
+        if not _is_allowed_by_moderation(description, context):
+            raise SteamshipError(
+                "Supplied 'description' was rejected by game's moderation filter. Please try again."
+            )
+
+        game_state = get_game_state(context)
+        game_state.player.description = description
+
+        if game_state.player.description and game_state.player.name:
+            if (not game_state.image_generation_requested()) or update:
+                if image_gen := get_image_generator(context):
+                    task = image_gen.request_profile_image_generation(context=context)
+                    character_image_block = task.wait().blocks[0]
+                    game_state.player.profile_image_url = (
+                        character_image_block.raw_data_url
+                    )
+                    game_state.profile_image_url = character_image_block.raw_data_url
+                    save_game_state(game_state, context)
```

Comment:
quibble: This line is effectively a dupe, right?  The save_game_state at the end of the method will always be executed next?